sky/provision/oci/query_utils.py

```diff
@@ -472,44 +474,113 @@ def delete_vcn(cls, net_client, skypilot_vcn, skypilot_subnet,
 
     @classmethod
     @debug_enabled(logger)
-    def add_ingress_rules(cls, region: str, ports: Set[int]):
-        if ports is None or len(ports) == 0:
+    def find_or_create_nsg(cls, region: str, cluster_name: str) -> str:
```

Comment:
```suggestion
    def find_or_create_nsg(cls, region: str, cluster_name: str) -> str:
```
accepting a "cluster name" in this function feels a little bit leaking abstraction to me. Should we instead pass name of the nsg here, and construct the name in the caller?